Mathlib/Topology/ClopenBox.lean

```diff
@@ -0,0 +1,182 @@
+/-
+Copyright (c) 2023 Dagur Asgeirsson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dagur Asgeirsson
+-/
+import Mathlib.Topology.Separation
+/-!
+
+# Clopen subsets in cartesian products
+
+In general, a clopen subset in a cartesian product of topological spaces cannot be written as a
+union of "clopen boxes", i.e. products of clopen subsets of the components (see
+[buzyakovaClopenBox] for counterexamples).
+
+However, when one of the components is compact, a clopen subset can be written as such a union.
+Our argument in `exists_clopen_box` follows the one given in [buzyakovaClopenBox].
+
+We deduce that in a product of compact spaces, a clopen subset is a finite union of clopen boxes,
+and use that to prove that the property of having finitely many clopens is preserved by taking
```

Comment:
Do you mean "finitely many" or "countably many" (which is what you show at the end of the file, and which seems more related to the "light" story)?